### PR TITLE
operators [CI] couchdb-operator (2.0.1)

### PR DESCRIPTION
**Pull request by welshDoug**
Thanks submitting your Operator. Please check below list before you create your Pull Request.

### New Submissions

* [ ] Are you familiar with our [contribution guidelines](https://github.com/operator-framework/community-operators/blob/master/docs/contributing-via-pr.md)?
* [ ] Have you [packaged and deployed](https://github.com/operator-framework/community-operators/blob/master/docs/testing-operators.md) your Operator for Operator Framework?
* [ ] Have you tested your Operator with all Custom Resource Definitions?
* [ ] Have you tested your Operator in all supported [installation modes](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/building-your-csv.md#operator-metadata)?
* [ ] Have you considered whether you want use [semantic versioning order](https://github.com/operator-framework/community-operators/blob/master/docs/operator-ci-yaml.md#semver-mode)?
* [ ] Is your submission [signed](https://github.com/operator-framework/community-operators/blob/master/docs/contributing-prerequisites.md#sign-your-work)?
* [ ] Is operator [icon](https://github.com/operator-framework/community-operators/blob/master/docs/packaging-operator.md#operator-icon) set?

### Updates to existing Operators

* [x] Did you create a `ci.yaml` file according to the [update instructions](https://github.com/operator-framework/community-operators/blob/master/docs/operator-ci-yaml.md)?
* [ ] Is your new CSV pointing to the previous version with the `replaces` property if you chose `replaces-mode` via the `updateGraph` property in `ci.yaml`? Only available in new channel
* [x] Is your new CSV referenced in the [appropriate channel](https://github.com/operator-framework/community-operators/blob/master/docs/packaging-operator.md#channels) defined in the `package.yaml` or `annotations.yaml` ?
* [x] Have you tested an update to your Operator when deployed via OLM?
* [x] Is your submission [signed](https://github.com/operator-framework/community-operators/blob/master/docs/contributing-prerequisites.md#sign-your-work)?

### Your submission should not

* [ ] Modify more than one operator
* [ ] Modify an Operator you don't own
* [ ] Rename an operator - please remove and add with a different name instead
* [ ] Modify any files outside the above mentioned folders
* [ ] Contain more than one commit. **Please squash your commits.**

### Operator Description must contain (in order)

1. [x] Description about the managed Application and where to find more information
2. [x] Features and capabilities of your Operator and how to use it
3. [x] Any manual steps about potential pre-requisites for using your Operator

### Operator Metadata should contain

* [x] Human readable name and 1-liner description about your Operator
* [x] Valid [category name](https://github.com/operator-framework/community-operators/blob/master/docs/packaging-operator.md#categories)<sup>1</sup>
* [x] One of the pre-defined [capability levels](https://github.com/operator-framework/operator-courier/blob/4d1a25d2c8d52f7de6297ec18d8afd6521236aa2/operatorcourier/validate.py#L556)<sup>2</sup>
* [x] Links to the maintainer, source code and documentation
* [x] Example templates for all Custom Resource Definitions intended to be used
* [x] A quadratic logo

Remember that you can preview your CSV [here](https://operatorhub.io/preview).

--

<sup>1</sup> If you feel your Operator does not fit any of the pre-defined categories, file an issue against this repo and explain your need

<sup>2</sup> For more information see [here](https://sdk.operatorframework.io/docs/overview/#operator-capability-level)

### CODE REVIEW

**Comment by welshDoug** [comment]
Is someone able to assist with understanding the errors? They suggest a problem that is larger than the bundle added in this PR

**Comment by mvalarh** [comment]
Lets retest it

**Comment by kasserater** [comment]
@mvalarh still no dice
what is the nature of the failure?

**Comment by camilamacedo86** [comment]
Hi welshDoug, 

The problem here is that you will break your upgrade graph. See that https://github.com/k8s-operatorhub/community-operators/blob/d762d9ed354ea242f8233503cb380fc2667fb734/operators/couchdb-operator/2.0.1/manifests/couchdb-operator.v2.0.1.clusterserviceversion.yaml is missing the config to let the catalog knows how to upgrade the distribution.

You need to use skips, skipRange or replaces in the CSV. See:  https://v0-18-z.olm.operatorframework.io/docs/concepts/olm-architecture/operator-catalog/creating-an-update-graph/

File: operators/couchdb-operator/2.0.1/bundle.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+LABEL operators.operatorframework.io.bundle.mediatype.v1=registry+v1
+LABEL operators.operatorframework.io.bundle.manifests.v1=manifests/
+LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
+LABEL operators.operatorframework.io.bundle.package.v1=couchdb-operator
+LABEL operators.operatorframework.io.bundle.channels.v1=stable,v2.1  
+LABEL operators.operatorframework.io.bundle.channel.default.v1=v2.1
+
+COPY /manifests /manifests/
+COPY /metadata /metadata/
```
Comment (camilamacedo86) [inline operators/couchdb-operator/2.0.1/bundle.Dockerfile:11]: This file will be igonored

File: operators/couchdb-operator/2.0.1/manifests/couchdb-operator.v2.0.1.clusterserviceversion.yaml [hunk 2]
```diff
@@ -91,7 +91,6 @@ metadata:
   namespace: placeholder
 spec:
   displayName: Operator for Apache CouchDB
-  replaces: couchdb-operator.v1.4.4
```
Comment (camilamacedo86) [inline operators/couchdb-operator/2.0.1/manifests/couchdb-operator.v2.0.1.clusterserviceversion.yaml]: Why are you trying to remve the replaces?

File: operators/couchdb-operator/2.0.1/metadata/annotations.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+annotations:
+  operators.operatorframework.io.bundle.channel.default.v1: v2.1
+  operators.operatorframework.io.bundle.channels.v1: stable,v2.1
```
Comment (camilamacedo86) [inline operators/couchdb-operator/2.0.1/metadata/annotations.yaml:3]: if you want to publish it only on v2.1 then you cannot add stable here. 
```suggestion
   operators.operatorframework.io.bundle.channels.v1: v2.1  
```

**Comment by kasserater** [comment]
> Hi welshDoug,
> 
> The problem here is that you will break your upgrade graph. See that https://github.com/k8s-operatorhub/community-operators/blob/d762d9ed354ea242f8233503cb380fc2667fb734/operators/couchdb-operator/2.0.1/manifests/couchdb-operator.v2.0.1.clusterserviceversion.yaml is missing the config to let the catalog knows how to upgrade the distribution.
> 
> You need to use skips, skipRange or replaces in the CSV. See: https://v0-18-z.olm.operatorframework.io/docs/concepts/olm-architecture/operator-catalog/creating-an-update-graph/

@camilamacedo86 we have restored the "replaces" but tests still fail.
can you please assist?

just to make sure our intent is clear: we want previous releases to upgrade to 2.0.1, but also we want there to be a new channel starting at 2.0.1
the reason? since 2.0.1 removes support for older CouchDB releases, we want the option for users of the operator to not have to upgrade if they don't want to

**Review by kasserater — APPROVED** [review]

**Comment by kasserater** [comment]
@mvalarh can you please assist with review of @welshDoug's commit and maybe also take a look at why the tests are still failing?

**Comment by kasserater** [comment]
@camilamacedo86, @J0zi, @mvalarh  please assist

**Pull request closed without merging**